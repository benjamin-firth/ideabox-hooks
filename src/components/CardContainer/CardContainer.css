.cards-container {
  display: flex;
  flex-wrap: wrap;
  height: 70vh;
  overflow-y: scroll;
}