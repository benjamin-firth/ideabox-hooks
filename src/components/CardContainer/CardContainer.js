import React from 'react';
import './CardContainer.css';
import Card from '../Card/Card';

const CardContainer = ({ ogReservations, deleteReservation }) => {
  const cards = ogReservations.map(card => {
    return(
      <Card 
        id={card.id} 
        name={card.name} 
        date={card.date} 
        time={card.time} 
        number={card.number} 
        deleteReservation={deleteReservation}
      />
    )
  })

  return(
    <div className="cards-container">
      {cards}
    </div>
  )
}

export default CardContainer;