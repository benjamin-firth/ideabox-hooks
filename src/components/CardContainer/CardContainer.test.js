import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer';

describe('CardContainer', () => {
  let wrapper;
  let mockOgReservations;


  beforeEach(() => {
    mockOgReservations = [
      {
        id: 123,
        name: 'test-ben',
        date: '12/12',
        time: '7:00',
        number: 3
      }
    ]
    wrapper = shallow(<CardContainer ogReservations={mockOgReservations} />)
  })

  describe('CardContainer Snapshots', () => {
    it('Should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })
  })

})