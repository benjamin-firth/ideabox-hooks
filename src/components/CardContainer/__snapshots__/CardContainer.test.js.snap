// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer CardContainer Snapshots Should match the snapshot 1`] = `
<div
  className="cards-container"
>
  <Card
    date="12/12"
    id={123}
    name="test-ben"
    number={3}
    time="7:00"
  />
</div>
`;
